### ISSUE TITLE
Upgrade aws-cdk-lib version 2.91.0 to 2.93.0

### ISSUE THREAD

**Issue by insik-han** [sst/sst#3263]
Hi.
I received mail from AWS.

```
Hello,

We are contacting you as we have identified that your AWS Account currently has one or more Lambda functions using the Node.js 14 runtime.

We are ending support for Node.js 14 in AWS Lambda. This follows Node.js 14 End-Of-Life (EOL) reached on April 30, 2023 [1].

As described in the Lambda runtime support policy [2], end of support for language runtimes in Lambda happens in two stages. Starting November 27, 2023, Lambda will no longer apply security patches and other updates to the Node.js 14 runtime used by Lambda functions, and functions using Node.js 14 will no longer be eligible for technical support. In addition, you will no longer be able to create new Lambda functions using the Node.js 14 runtime. Starting January 25, 2024, you will no longer be able to update existing functions using the Node.js 14 runtime.

We recommend that you upgrade your existing Node.js 14 functions to Node.js 18 before November 27, 2023.

End of support does not impact function execution. Your functions will continue to run. However, they will be running on an unsupported runtime which is no longer maintained or patched by the AWS Lambda team.

This notification is generated for functions using the Node.js 14 runtime for the $LATEST function version. For a list of your affected Lambda functions, please see the 'Affected resources" tab of your AWS Health Dashboard

The following command shows how to use the AWS CLI [3] to list all functions in a specific region using Node.js 14, including published function versions. To find all such functions in your account, repeat this command for each region:

aws lambda list-functions --function-version ALL --region us-east-1 --output text --query "Functions[?Runtime=='nodejs14.x'].FunctionArn"

If you have any concerns or require further assistance, please contact AWS Support [4].

[1] https://endoflife.date/nodejs
[2] https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html
[3] https://aws.amazon.com/cli/
[4] https://aws.amazon.com/support

Sincerely,
Amazon Web Services
```

Currently, sst use aws-cdk-lib@2.91.0 that use node.js 14.x when create certificateManager. [link](https://github.com/aws/aws-cdk/issues/25335)

So, I think sst should upgrade aws-cdk-lib@2.91.0 to 2.93.0

**Comment by mng-prophero** [comment]
We're having the same issue 

```
"node_modules/sst": {
			"version": "2.24.20",
			"resolved": "https://registry.npmjs.org/sst/-/sst-2.24.20.tgz",
			"integrity": "sha512-o+FkY8DierOY3+/zvL4PbeCI/0CEK6e2m0amEMPVoYlq37ajPwjXDjRCC5ZZlV9PKmUHwHHyXF99PJ1wruah/A==",
			"dev": true,
			"dependencies": {
				"@aws-cdk/aws-apigatewayv2-alpha": "^2.91.0-alpha.0",
				"@aws-cdk/aws-apigatewayv2-authorizers-alpha": "^2.91.0-alpha.0",
				"@aws-cdk/aws-apigatewayv2-integrations-alpha": "^2.91.0-alpha.0",
				"@aws-cdk/cloud-assembly-schema": "2.91.0",
				"@aws-cdk/cloudformation-diff": "2.91.0",
				"@aws-cdk/cx-api": "2.91.0",
				"@aws-crypto/sha256-js": "^5.0.0",
				"@aws-sdk/client-cloudformation": "^3.279.0",
				"@aws-sdk/client-ecs": "^3.279.0",
				"@aws-sdk/client-eventbridge": "^3.342.0",
				"@aws-sdk/client-iam": "^3.279.0",
				"@aws-sdk/client-iot": "^3.279.0",
				"@aws-sdk/client-iot-data-plane": "^3.279.0",
				"@aws-sdk/client-lambda": "^3.279.0",
				"@aws-sdk/client-rds-data": "^3.279.0",
				"@aws-sdk/client-s3": "^3.279.0",
				"@aws-sdk/client-ssm": "^3.279.0",
				"@aws-sdk/client-sts": "^3.279.0",
				"@aws-sdk/config-resolver": "^3.272.0",
				"@aws-sdk/credential-providers": "^3.279.0",
				"@aws-sdk/middleware-retry": "^3.272.0",
				"@aws-sdk/middleware-signing": "^3.272.0",
				"@aws-sdk/signature-v4-crt": "^3.272.0",
				"@aws-sdk/smithy-client": "^3.279.0",
				"@babel/core": "^7.0.0-0",
				"@babel/generator": "^7.20.5",
				"@babel/plugin-syntax-typescript": "^7.21.4",
				"@smithy/signature-v4": "^2.0.4",
				"@trpc/server": "9.16.0",
				"adm-zip": "^0.5.10",
				"aws-cdk-lib": "2.91.0",
```

**Comment by jayair** [comment]
Just to double check, does 2.93 contain the version bump?

**Comment by TimQuist** [comment]
Did a deploy of a CDK app using '2.93.0' and my lambda's are now running Node.js 18. 

You can check the default in the 'aws-cdk-lib' -> 'aws-lambda-nodejs' -> 'lib' -> 'function.d.ts'
![Screenshot 2023-09-01 at 12 10 35](https://github.com/sst/sst/assets/47602036/860c74a9-abc3-46ff-bc96-eb1d007d987a)

**Comment by mixman** [comment]
When updating awsk-cdk-lib, latest v2.94.0 contains a fix for LogRetention errors on deployment https://discord.com/channels/983865673656705025/1144283085714763797/1144283085714763797

**Comment by mixman** [comment]
And v2.95.0 has further logRetention related fixes.
- logs: log retention custom resource timed out during deploy

**Comment by fwang** [comment]
Fixed in v2.25.0
